doc/nrf/libraries/others/nrf_compression.rst

```diff
@@ -0,0 +1,204 @@
+* If a library only supports decompression, the two decompression functions must be defined, and the compression function must be set to ``NULL``.
+* If a library supports both compression and decompression, all three functions must be defined.
+
+All other functions are always mandatory and must always be defined.
+Additionally, you must define a unique ID for the compression library.
+It should be globally available, allowing applications or modules use it.
+
+Using the compression subsystem
+*******************************
+
+To decompress data using the nRF Compression library, complete the following steps:
+
+1. Use the :c:func:`nrf_compress_implementation_find` function with the ID of the desired compression type.
+   If it returns ``NULL``, the compression type is not supported.
+#. Check the ``init``, ``deinit`` and ``reset`` core function pointers.
+   If any return ``NULL``, there is an implementation issue with the compression type.
+#. For decompression support, check the :c:type:`nrf_compress_decompress_bytes_needed_t` and :c:func:`nrf_compress_decompress_func_t` function pointers.
+   If either is ``NULL``, there is an implementation issue with the compression type.
+#. Call the :c:func:`nrf_compress_init_func_t` function to set up the compression library.
+   If a non-zero error is returned, the library cannot be initialized, possibly due to insufficient memory or other issues.
+#. Call the :c:type:`nrf_compress_decompress_bytes_needed_t` function to determine how many bytes of data should be specified for decompression.
+   If a non-zero error is returned, there is an issue with the compression library.
+#. Call the :c:func:`nrf_compress_decompress_func_t` function with the requested amount of data (or the maximum available, if less is available).
+   In case there is no more data to process, set the ``last_part`` value to true, otherwise set it to false.
+   If a non-zero error is returned, the provided data might be invalid.
+#. Ensure that if the ``offset`` updated value is not equal to the size of the input data provided, the next call to  :c:func:`nrf_compress_decompress_func_t` function includes the unused bytes from the buffer at its start.
+#. Verify if the ``output_size`` value is greater than ``0``.
+   In such a case, the specified amount of data is available in the ``output`` buffer and you should copy it or move it to its intended destination.
+#. Repeat the process of calling the :c:type:`nrf_compress_decompress_bytes_needed_t` function followed by  :c:func:`nrf_compress_decompress_func_t` until all the data has been processed.
+#. Call the :c:func:`nrf_compress_deinit_func_t` function to clean up the compression library.
+
+.. figure:: nrf_compression_image.png
+    :alt: nRF compress library decompression flowchart
+
+
+
+
+
+MCUboot image compression
+#########################
```

Comment:
This header level is only for page titles. I suggest the following:
```suggestion
*************************
```
And then changing the following subsection underlines accordingly to one level lower.

Reply:
Should clarify that this will be a different page, I've got it in one file at present just to see it in the documentation output but it will be moved to a different file once finished